From aa6f9f9b99d85f4c60dad78f22a201edd295b1cb Mon Sep 17 00:00:00 2001
From: David Sernelius <david.sernelius@flir.se>
Date: Fri, 15 Nov 2019 06:34:57 +0100
Subject: [PATCH] Set standby state of camera pins also when not streaming

The standby pin configuration of the vcam module was not set when not streaming, with this fix it should be set all the time.
---
 drivers/media/platform/mxc/capture/mxc_viu.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/mxc_viu.c b/drivers/media/platform/mxc/capture/mxc_viu.c
index 297a5e7e876f..d5d8b06b353e 100644
--- a/drivers/media/platform/mxc/capture/mxc_viu.c
+++ b/drivers/media/platform/mxc/capture/mxc_viu.c
@@ -1522,7 +1522,7 @@ static int imx_viu_suspend(struct device *dev)
 	pr_info("imx_viu_suspend\n");
 
 	if (!vb2_is_streaming(&viu_dev->queue))
-		return 0;
+		goto pin_sleep;
 
 	ret = wait_for_completion_timeout(&viu_dev->dma_done, HZ / 10);
 	if (!ret) {
@@ -1532,14 +1532,15 @@ static int imx_viu_suspend(struct device *dev)
 	imx_viu_irq_disable(viu_dev, ERROR_IRQ);
 	imx_viu_irq_disable(viu_dev, DMA_END_IRQ);
 
-	pinctrl_pm_select_sleep_state(dev);
-
 	/* save registers for resume */
 	imx_viu_save_reg_stack(viu_dev, &viu_dev->reset);
 
 	clk_disable_unprepare(viu_dev->ipg_clk);
 	release_bus_freq(BUS_FREQ_HIGH);
 
+pin_sleep:
+	pinctrl_pm_select_sleep_state(dev);
+
 	return 0;
 }
 
@@ -1551,7 +1552,7 @@ static int imx_viu_resume(struct device *dev)
 	pr_info("imx_viu_resume\n");
 
 	if (!vb2_is_streaming(&viu_dev->queue))
-		return 0;
+		goto pin_restore;
 
 	request_bus_freq(BUS_FREQ_HIGH);
 	clk_prepare_enable(viu_dev->ipg_clk);
@@ -1561,6 +1562,7 @@ static int imx_viu_resume(struct device *dev)
 	imx_viu_irq_enable(viu_dev, DMA_END_IRQ);
 	imx_viu_irq_enable(viu_dev, ERROR_IRQ);
 
+pin_restore:
 	pinctrl_pm_select_default_state(dev);
 
 	return 0;
-- 
2.17.1

